src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -272,11 +275,25 @@ public void execute()
 
     private boolean projectHasAlreadySetAnArtifact()
     {
-        final File artifact = getProject().getArtifact().getFile();
-        if ( artifact != null )
+        if ( getProject().getArtifact().getFile() != null )
         {
-            final String defaultArtifactName = artifact.getName().replace( ".jar", "" );
-            return artifact.isFile() && defaultArtifactName.equals( finalName );
+            if ( getProject().getArtifact().getFile().isFile() )
+            {
+                try
+                {
+                    JarFile existingJarFile = new JarFile( getProject().getArtifact().getFile() );
+                    Attributes existingJarFileAttr = existingJarFile.getManifest().getMainAttributes();
```

Comment:
Sorry for confusing you. I've only added the check with the MANIFEST file in the integration test to make it succeed on to explain how to verify resources in a jar.
Back to my biggest concern: I'm afraid that if users play with final names that the check it not good and might install the wrong artifact. I want to ensure that the first one (no matter the final name) always wins. Using the same classifier with different final names should also lead to an exception.
I hope this makes it a bit more clear.

Reply:
OK, I got it. I guess were are wasting our time then. From this [PR #1 - Fixes filename conflict when producing 2 jars without classifier](https://github.com/apache/maven-jar-plugin/pull/1/files), the test was already creating two artifacts with different names and the same classifier. The correct behavior already happens and the first JAR is already the one that is installed.

This test case is invalid and I can't do anything about it. As in [https://issues.apache.org/jira/browse/MJAR-254], it is stated that:

> In summary, the build should fail if the actual JAR file would get overridden, not if the classifier happens to be the same. That is, the conflict is at the actual filesystem level.

but this is a behavior that should be invalid.

I am not sure what to do now.

Reply:
So you are saying: there is actually no issue? I have always wondered why the filename mattered... does it really matter?

Reply:
If I understood the issue well, no.

> Using the same classifier with different final names should also lead to an exception.

The test case provided was testing exactly this situation.

> So you are saying: there is actually no issue? I have always wondered why the filename mattered... does it really matter?

I guess no. Since all the generated artifacts would have different classifiers, when 'blank' is one possible value for the classifier, I don't think so.

From all I got here, I'm totally open for being contested and suggestions. I might be wrong and any number of opinions would be good for me. Someone else should have a better grasp of this issue better than I have.

Reply:
@rfscholte Is it possible to create a finalName = <Name of the artifact without> classifier and setting a classifier on it, just to circumvent maven-jar-plugin and override the original JAR?

Example:
A-1.0-SNAPSHOT.jar
A-1.0-SNAPSHOT-cl1.jar -> and in pom.xml I set A-1.0-SNAPSHOT-cl1 to be A-1.0-SNAPSHOT

Reply:
Within the maven-jar-plugin that's not possible to do. classifier is always respected, finalName doesn't matter, during install/deploy it will use the artifactId anyway.

Reply:
@rfscholte Is the case to reject the issue and ignore this pull request? I'm fine with whatever resolution. Is it something worthy to ask Maven dev mailing list?

Reply:
I've added a comment to MJAR-254. If we don't get good feedback, I'll close it.